frontend/apps/web/components/site-header/Upgrade.tsx

```diff
@@ -94,15 +131,95 @@ function UpgradeInfoDialog(props: UpgradeInfoDialogProps): ReactElement {
 }
 
 interface IncludedReasonProps {
-  reason: string;
+  reason: AccountStatus;
+  description: string;
+  count: number;
+  allowedRecords: number;
 }
 
 function IncludedReason(props: IncludedReasonProps): ReactElement {
-  const { reason } = props;
+  const { reason, description, count, allowedRecords } = props;
+
+  switch (reason) {
+    case AccountStatus.EXCEEDS_ALLOWED_LIMIT:
+      return (
+        <UsageLimitExceeded
+          current={count}
+          allowed={allowedRecords}
+          title={'Usage Limit Exceeded'}
+          description={`You have used all of your allowed records for this billing cycle. `}
+        />
+      );
+    case AccountStatus.REQUESTED_EXCEEDS_LIMIT:
+      return (
+        <UsageLimitExceeded
+          current={count}
+          allowed={allowedRecords}
+          title={'Usage Limit Warning'}
+          description={`The requested record count woudl exceed your alowed records for this billing cycle. `}
+        />
+      );
+    case AccountStatus.ACCOUNT_IN_EXPIRED_STATE:
+      return (
+        <Alert>
+          <IoAlertCircleOutline className="h-4 w-4" />
+          <AlertTitle>Account is Expired</AlertTitle>
+          <AlertDescription>{description}</AlertDescription>
+        </Alert>
+      );
+    case AccountStatus.REASON_UNSPECIFIED:
```

Comment:
I think we updated the thinking that if this is unspecified, the account is actually in a valid state.